soh/soh/UIWidgets.hpp

```diff
@@ -56,12 +57,13 @@ namespace UIWidgets {
     void PaddedSeparator(bool padTop = true, bool padBottom = true, float extraVerticalTopPadding = 0.0f, float extraVerticalBottomPadding = 0.0f);
     bool EnhancementCheckbox(const char* text, const char* cvarName, bool disabled = false, const char* disabledTooltipText = "", CheckboxGraphics disabledGraphic = CheckboxGraphics::Cross, bool defaultValue = false);
     bool PaddedEnhancementCheckbox(const char* text, const char* cvarName, bool padTop = true, bool padBottom = true, bool disabled = false, const char* disabledTooltipText = "", CheckboxGraphics disabledGraphic = CheckboxGraphics::Cross, bool defaultValue = false);
-    void EnhancementCombo(const std::string& name, const char* cvarName, const std::vector<std::string>& items, int defaultValue = 0);
-    bool EnhancementCombobox(const char* name, const char* ComboArray[], size_t arraySize, uint8_t FirstTimeValue, bool disabled = false, const char* disabledTooltipText = "", uint8_t disabledValue = -1);
+    bool EnhancementCombobox(const char* cvarName, std::span<const char*, std::dynamic_extent> comboArray, uint8_t defaultIndex, bool disabled = false, const char* disabledTooltipText = "", uint8_t disabledValue = -1);
+    bool LabeledRightAlignedEnhancementCombobox(const char* label, const char* cvarName, std::span<const char*, std::dynamic_extent> comboArray, uint8_t defaultIndex, bool disabled = false, const char* disabledTooltipText = "", uint8_t disabledValue = -1);
```

Comment:
i'd like to hear what others think about the switch to using `std::span` here, i know we have a pattern of using the `ARRAY_COUNT` macros throughout the rest of the codebase

https://github.com/HarbourMasters/Shipwright/blob/dbcc6b4818c9c23791effd2787fa5b180a8d7542/soh/include/macros.h#L20-L21

i don't have strong feelings one way or the other, i know `ARRAY_COUNT` works in c code so using that makes sense if the goal is consistent patterns throughout the codebase, but `std::span` makes sense from a "don't reinvent the wheel" and "use the tools available" stand point

Reply:
`std::span` is C++20, and I've heard mixed feelings from the team about using higher standard library features. It would help to know what version we are comfortable supporting up to, and have that defined somewhere (and eventually cleanup any usages that are out of the supported range).

Reply:
Gonna tag in @dcvz and @aMannus in that case.  The main problem that using this solves is that now we don't need to pass in the array size as a separate parameter at all.  Passing the array sizes separately is very error prone, especially as people modify settings in the future.  Using span finally brings C++ up to parity with other languages that actually know how big their own arrays are.  

The biggest implications of this change are in the randomizer code, which was using MAX entries for every single option enum _only_ for drawing the comboboxes.  That practice is also error-prone, and honestly kind of ridiculous since most of those options only have a few choices.  The MAX entries make sense for enums with tons of options, but in the randomizer, they were just being used to get around C++'s deficiencies.  

The only drawback this approach has is in the couple of places where people were using the same array of options in multiple comboboxes but just a subset of them, in which case I had to create separate arrays of pointers, which is still relatively cheap compared to duplicating the strings themselves.  Also, removing the array size argument from the combobox function and using the sizeof/sizeof approach inside the function doesn't work due to pointer decaying.

To answer @Archez, obviously this compiles as-is.  We are already building with standard C++20 support, so while I understand the philosophical argument, I'd love to know if there is any technical argument backing it up.  If so, it probably doesn't make sense to keeping building with C++20 at all.  I know there's been talk about specific _platform_ features not being supported on all platforms or being platform-dependent, but this _language_ feature seems like it should be supported everywhere.

Reply:
I'm not too familiar with the differences in C++ version or what it means for supporting different platforms. If it doesn't get in the way of that, I don't see why not.